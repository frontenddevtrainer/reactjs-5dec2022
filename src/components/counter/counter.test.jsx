import { render, screen } from "@testing-library/react";
import { act } from "react-dom/test-utils";
import Counter from "./counter";

describe("Unit test for: Counter Component", ()=>{
    test("Counter component should render", ()=>{
        render(<Counter/>);
    })

    test.todo("Increase counter value")

    test("Load counter with 0", ()=>{
        render(<Counter/>);
        const counterValue = screen.getByTestId("counter_value_text");
        expect(counterValue.textContent).toEqual("0");

    })

    test("Decrease counter value", ()=>{
        render(<Counter/>);

        const decreaseButton = screen.getByTestId("counter_decrease_button");
        const counterValue = screen.getByTestId("counter_value_text");

       act(()=>{
        decreaseButton.click()
       })

        expect(counterValue.textContent).toEqual("-1");

    })

    test("Increse counter value", ()=>{
        render(<Counter/>);

        const increaseButton = screen.getByTestId("counter_increase_button");
        const counterValue = screen.getByTestId("counter_value_text");

       act(()=>{
        increaseButton.click()
       })

        expect(counterValue.textContent).toEqual("1");

    })

    test("Snaopshot test", ()=>{
        const { asFragment } = render(<Counter/>);
        const content = asFragment();
        expect(content).toMatchSnapshot();
    })

    test("Snaopshot test after increasing the value", ()=>{
        const { asFragment } = render(<Counter/>);
        const increaseButton = screen.getByTestId("counter_increase_button");
        act(()=>{
            increaseButton.click()
           })
        const content = asFragment();
        expect(content).toMatchSnapshot();
    })

    test("Snaopshot test after decreasing the value", ()=>{
        const { asFragment } = render(<Counter/>);
        const increaseButton = screen.getByTestId("counter_decrease_button");
        act(()=>{
            increaseButton.click()
           })
        const content = asFragment();
        expect(content).toMatchSnapshot();
    })

})