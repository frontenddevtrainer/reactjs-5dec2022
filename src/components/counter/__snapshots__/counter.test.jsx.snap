// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for: Counter Component Snaopshot test 1`] = `
<DocumentFragment>
  <div>
    <button
      data-testid="counter_decrease_button"
    >
      -
    </button>
    <strong
      data-testid="counter_value_text"
      style="color: red;"
    >
      0
    </strong>
    <button
      data-testid="counter_increase_button"
    >
      +
    </button>
    <p>
      Current time: 
    </p>
  </div>
</DocumentFragment>
`;

exports[`Unit test for: Counter Component Snaopshot test after decreasing the value 1`] = `
<DocumentFragment>
  <div>
    <button
      data-testid="counter_decrease_button"
    >
      -
    </button>
    <strong
      data-testid="counter_value_text"
      style="color: blue;"
    >
      -1
    </strong>
    <button
      data-testid="counter_increase_button"
    >
      +
    </button>
    <p>
      Current time: 
    </p>
  </div>
</DocumentFragment>
`;

exports[`Unit test for: Counter Component Snaopshot test after increasing the value 1`] = `
<DocumentFragment>
  <div>
    <button
      data-testid="counter_decrease_button"
    >
      -
    </button>
    <strong
      data-testid="counter_value_text"
      style="color: blue;"
    >
      1
    </strong>
    <button
      data-testid="counter_increase_button"
    >
      +
    </button>
    <p>
      Current time: 
    </p>
  </div>
</DocumentFragment>
`;
